content/vagrant-local-for-k8s.md

```diff
@@ -0,0 +1,193 @@
+    zypper --non-interactive in gnu-netcat curl
+    curl -sfL https://get.k3s.io | sh -
+    export KUBECONFIG=/etc/rancher/k3s/k3s.yaml
+    cp /etc/rancher/k3s/k3s.yaml /vagrant/
+  SHELL
+
+# Vagrant Cloud and a few tips for choosing your base image
+
+The base image you declare via `config.vm.box` in the vagrant file comes from a cloud service called Vagrant Cloud, which can be found at [https://app.vagrantup.com](https://app.vagrantup.com/boxes/search). Essentially, Vagrant Cloud is for VMs what docker hub is for containers - a salad bowl of all sorts of VM images. Some of them are generic base boxes, some of them are tailored towards specific use cases by means of specific configuration and selection (addition and removal) of software contained in the image. Of course you'll find boxes containing the base operating systems of all major Linux distributions such as Ubuntu, CentOS and, of course openSUSE. 
+
+It is important to note that the same distro and version are sometimes provided by more than one account. Everybody can create an account on vagrant cloud and name it whatever they want, unless the name isn't taken yet. For example, the official vagrant images maintained by the openSUSE community can be found under [https://app.vagrantup.com/opensuse](https://app.vagrantup.com/opensuse). 
+
+However, if you just search for "opensuse", your first hit will likely be an image called `generic/opensuse15`, which happens to show up first in the list because it has more downloads than the official image. This image is done by the [https://roboxes.org](https://roboxes.org) project, which provides a large variety of virtual machine images built with Hashicorp's [packer](https://www.packer.io) tool. Certainly a credible source too, but, as far as I am concerned, I'd rather go for the original. Particularly since it's not entirely clear which openSUSE [flavor](https://software.opensuse.org) this image actually contains - there's openSUSE Leap, Tumbleweed, Kubic, etc. By inference I would guess the `generic/opensuse15` is based on openSUSE Leap, since Leap's current major version is indeed 15, but is it 15.0, 15.1, or 15.2? Too much headache for my taste...
+
+Each image is specific to a certain so-called provider, which essentially means hypervisor. So if you're using for example Virtualbox as the underlying virtualization provider, you need to make sure your chosen image supports that provider. One neat feature of vagrant cloud is that there are two little tabs on each image's page titles "vagrantfile" and "new". Vagrantfile gives you the lines you need to copy paste into your - you guessed it - vagrantfile to use this image, and the new tab gives you the commands to init and spin a box up with this image in one go.
+
+# Managing the local image cache
+
+When you, as we did in our example above, specify the vagrant image without declaring a specific version like
+
+   config.vm.box = "opensuse/Leap-15.2.x86_64"
```

Comment:
Can you run aarch64 VMs in a x86 environment? Do you happen to know if this is affected by the move to M1 for Apple?

Reply:
I think so? depends if VB supports it but in principle it should work.